shared/actions/chat/inbox.js

```diff
@@ -66,10 +65,9 @@ function * _backgroundUnboxLoop () {
 
 // Update inboxes that have been reset
 function * _updateFinalized (inbox: ChatTypes.GetInboxLocalRes) {
-  const finalizedState: Constants.FinalizedState = Map((inbox.conversationsUnverified || []).map(convoUnverified => [
-    Constants.conversationIDToKey(convoUnverified.metadata.conversationID),
-    convoUnverified.metadata.finalizeInfo,
-  ]))
+  const finalizedState: Constants.FinalizedState = Map((inbox.conversationsUnverified || []).reduce((map, convoUnverified) => {
```

Comment:
```js
Immutable.Map([['foo', 123], ['bar', 456]]).toJS()
// => {foo: 123, bar: 456} Immutable.Map
```

It's more efficient than using reduce since you aren't constantly creating new Maps (even if they're sharing data)